frontend/src/app/main/ui/workspace/snap_points.cljs

```diff
@@ -65,11 +110,12 @@
 
         ;; We use sets to store points/lines so there are no points/lines repeated
         ;; can cause problems with react keys
-        snap-points (into #{} (mapcat (fn [[point snaps coord]]
-                                        (cons point snaps))
-                                      @state))
-        snap-lines (into #{} (mapcat (fn [[point snaps coord]]
-                                       (when (not-empty snaps) (map #(vector point %) snaps))) @state))]
+        snap-points (->> @state
+                         (mapcat add-point-to-snaps)
+                         (into #{}))
+
```

Comment:
You can avoid creating intermediate seqs with `(into #{} (mapcat add-point-to-snaps) @state)`

Reply:
OK